internal/validation/apiextensions/v1/xrd/handler.go

```diff
@@ -145,16 +146,18 @@ func (v *validator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.O
 }
 
 func (v *validator) dryRunUpdateOrCreateIfNotFound(ctx context.Context, crd *apiextv1.CustomResourceDefinition) error {
-	got := crd.DeepCopy()
-	err := v.client.Get(ctx, client.ObjectKey{Name: crd.Name}, got)
-	if err == nil {
-		got.Spec = crd.Spec
-		return v.client.Update(ctx, got, client.DryRunAll)
-	}
-	if kerrors.IsNotFound(err) {
-		return v.client.Create(ctx, crd, client.DryRunAll)
-	}
-	return err
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		got := crd.DeepCopy()
+		err := v.client.Get(ctx, client.ObjectKey{Name: crd.Name}, got)
+		if err == nil {
+			got.Spec = crd.Spec
+			return v.client.Update(ctx, got, client.DryRunAll)
+		}
+		if kerrors.IsNotFound(err) {
+			return v.client.Create(ctx, crd, client.DryRunAll)
+		}
```

Comment:
This can be outside of the loop.

Reply:
Yes, correct. Keeping it in the retry function is more concise though, as in the retry function we still need to try to get the CRD to retry an update.